templates/template-for-tasks.rst

```diff
@@ -0,0 +1,171 @@
+
+.. - For these subtasks, a table will be shown with 3 columns:
+
+..  - Subtask name
+..  - Default target
+..  - Description of what it does
+
+`Details about the Retargetable Subtasks Subsection  <writing-task-topics.html#task-topics-retarg>`_ .
+   
+Parameters
+----------
+
+.. This section will be autofilled also -- the content is filled in
+   from docstrings in the code itself, not in this reST document (see
+   Guidance Doc for details).
+
+.. Here, configuration parameters will be displayed in a table with the following fields:
+
+.. - Parameter name.
+
+.. - Parameter type.  These are generally simple python var types
+   (i.e. `bool`, `int`, `float`, or `str`) , which will automatically
+   be linked to existing python documentation on these types)
+
+.. - Default value of parameter.
+
+.. - A description sentence or paragraph. 
+
+`Details about the Parameters Subsection  <writing-task-topics.html#task-topics-params>`_ .
+
+Python usage
+============
+
+Class initialization
+--------------------
+
+..  This section will be autofilled also -- the content is filled in
+    from docstrings in the code itself, not in this reST document (see
+    Guidance Doc for details).
+
+.. This section consists of:
+
+.. - Interface for declaring an instance of the class
+  
+.. - Description of the parameters in the interface signature
+
+`Details about the Class initialization Subsection  <writing-task-topics.html#task-topics-initzn>`_ .
+
+Run method
+----------
+
+.. This section will be autofilled also -- the content is filled in
+   from docstrings in the code itself, not in this reST document (see
+   Guidance Doc for details).
+
+.. This will consist of:
+
+.. - A description of the interface for calling the primary entrypoint
+   function for the class -- again, this will be picked up
+   automatically from the interface of the `run` method and will not
+   require developer input.
+
+.. - A short description of what the `run` method requires as required and optional inputs
+
+.. - Description of the parameters in the run signature
+
+`Details about the Run Method Subsection  <writing-task-topics.html#task-topics-run>`_ .
+
+
+Debugging
+=========
+
+.. This section will be autofilled also -- the content is filled in
+   from docstrings in the code itself, not in this reST document (see
+   Guidance Doc for details).
+
+.. - Debugging framework hooks: if there are several debugging
+   parameters, they will be displayed in a table similar to how the
+   configuration parameters are done, with three columns:
+
+..  - Parameter name
+..  - Parameter type
+..  - Parameter description
+
+`Details about the Debugging Section  <writing-task-topics.html#task-topics-debug>`_ .
+    
+Examples
+========
+
+.. - Fill in a self-contained example of using this task that can be tested by any reader.
+
+`Guidance for the Examples Subsection  <writing-task-topics.html#task-topics-examples>`_ .
+   
+Algorithm details
+====================
```

Comment:
Yeah, sorry - i try to watch for these, but some slip through. 

(However, i used to get warnings when i did the build whenever this happened, but no longer do..  my conf.py is pretty minimal so i don't think there's anything in it which is suppressing the warnings, but something else must've changed.. (?) )